### ISSUE TITLE
[Ray Dashboard]  Set Route Prefix/Base Address

### ISSUE THREAD

**Issue by BrentSouza** [ray-project/ray#35269]
### Description

I think it would be nice to have the ability to set a base address for Ray Dashboard routes.

### Use case

We are hosting a number of different ray clusters behind an AWS ALB using path based routing.  ALBs don't support url rewriting so when we try to access the Dashboard, it forwards the path along, e.g. `/cluster-foo/dashboard`, which the dashboard server doesn't recognize as a valid route and returns `404: Not Found`. I imagine this could be solved if we could set a base address (similar to ray serve deployments `route prefix` or the `root_path` arg to `serve.start`).

**Comment by alanwguo** [comment]
makes sense

**Comment by alanwguo** [comment]
It would be quite costly to implement path_prefix beyond just the index.html file. If we are to prefix all the API endpoints as well, this would break CLI commands. 

@BrentSouza , do you/your ray users use the ray CLI commands against these ray clusters you are hosting?

**Comment by BrentSouza** [comment]
No. We do not use the cli currently in this use case. The clusters are running Ray Serve with several deployments and we're able to route the http requests accordingly using Ray Serve's support for controlling the path.
